### DISCUSSION TITLE
Messages visibility time is too short

### DISCUSSION THREAD

**Discussion by ANDREAPIPPI** [NightscoutFoundation/xDrip#2984] [General]
Hi.
Just an example: when I NFC scan a new Libre 2 sensor to start it,  and then rotate the phone to see the screen, I have just a short flash of a message, that disappears almost immediately.
And this happens for all messages, I think its time be set to no more than 5 seconds, may be even less.
Could it be possible to increase the message visibility time to 10 seconds, or something?
Thank you

**Comment by Navid200** [comment]
My understanding is that any toast message is accompanied by a log message.
The only reason for a toast is to get your attention.
You should be able to go to logs and see any important message even if it happened 2 hours ago let alone 2 minutes ago.

So, if this is not the case for any particular message, please let me know.

**Comment by ANDREAPIPPI** [comment]
Hi Navid.
Thanks, I was missing this information, good to know and use.
But this doesn't solve the problem: sometimes the toast message is not information, but a request for action, so I need to see this message right away, because going to the log and reading it will be too late to do anything.
I've been having this problem mainly when NFC scanning, so the screen is not visible, and after listening to the NFC sounds and rotating the phone to see the screen, the message is gone.
I assume there is a time set for screen toast messages so I'm asking to increase the duration to at least 10-12 seconds, if it's possible in a simple way, otherwise never mind.
Thank you.

**Reply by Navid200** [reply]
I need to see a toast message that requires a critical action.
I cannot do anything if I don't see it.

Even if you tell me what the requested action is, I can find it.

If there is a toast message that requires an action, I need to see it.

I can give you an example:
If you tap on stop sensor, a message comes on screen asking you to confirm.
The message never disappears.
Because it would make no sense if it did.  It is such a critical message.

So, I need to see what you are describing because without it, I have a hard time believing.

**Comment by ANDREAPIPPI** [comment]
Hi Navid.
I understand your doubts, and I'll try to shot such message at the next Sensor start, two weeks from now.
But as an example, in the "Libre 2 Direct" instructions here [https://xdrip.readthedocs.io/en/latest/install/libre2/#connect-to-the-sensor]
in the paragraph "**Starting Libre 2 Sensor**s", after stopping the sensor I read 
"**Scan the sensor with xDrip+ NFC feature and make sure you see the toast messages -Scanning..- & then -Scanned OK!**-."
Never had the time to see them....
Bye and thank you for the quick reply.
Andrea :-)

**Comment by Navid200** [comment]
@psonnera
We should have no critical message shown only as a toast message.  
Since I don't use Libre, I have no way to know exactly what is going on here.
Would you please share your experience and advice?

**Reply by psonnera** [reply]
@Navid200 these are toast messages related to L2 sensors sequence of pairing... as toast messages they have a limited timespan. Maybe @tzachi-dar can give his input? I'd rather see them on the top of the graph with other important messages (like invalid slope/intercept) but that might require some heavy rework.

**Reply by Navid200** [reply]
Can I have the English version of the message?

**Comment by ANDREAPIPPI** [comment]
I would be happy if timespan could just be al least 10-12 seconds, and probably this could be simpler to be implemented.

**Comment by ANDREAPIPPI** [comment]
Hello.
There are no comments here on this issue, but I think it's important, because it makes it impossible to do the right thing the right way.
This is a shame, because the topic deals with FSL 2, which is about to become the most widely used, instead of the FSL 1.
 @tzachi-dar, please, could you give your opinion/help on this problem?
Thanks a lot, appreciated.

**Comment by tzachi-dar** [comment]
After scanning the sensor xDrip log will have one of the following messages:
1) Bluetooth connection with sensor enabled, you should be able to receive BG data using Bluetooth.
2) Bluetooth connection with sensor failed. You wont get any readings. Try scanning again to fix this.

Look for this messages and you will know the status of the last scan.
If neither is printed, scan again.

**Comment by ANDREAPIPPI** [comment]
Thank you  tzachi-dar  for the quick reply.
The problem is with a FSL 2 Direct use through OOP2, when the sensor loses the BT connection.
Really I never had the possibility to read any of those messages, because as I NFC scan the sensor and rotate the phone to read the screen, they disappear immediately.
I had just a fleeting vision of a square black box with may be four scripts in the corners, and the xDrip+ logo on the right.
I supposed it was saying "scan under xDrip+" but if I do that the process starts again, in a loop.
It's very disappointing, but IMHO it's only a matter of time set to show the message, I think about 3-4 seconds, no more.
The strange thing is that if I do nothing after my first NFC scan, the BT connection goes ON, and everything works OK.
Strqange, isn't it?

**Comment by ANDREAPIPPI** [comment]
Hi
After a few attempts I managed to take a screenshot of the message that lasts too little.
This:
![553](https://github.com/NightscoutFoundation/xDrip/assets/32884006/dfb4bb6e-ebdf-4688-9f84-a7a6a41260fa)
As you can see, it gives you the option to choose between 2 actions, Youggluco or xDrip+ NFC.
But it is almost impossible to read it, due to the very short visibility on the screen.
Hope this may help...

**Reply by Navid200** [reply]
That cannot be an xDrip message.
It looks like an Android message.

**Reply by psonnera** [reply]
@Navid200 agree. @ANDREAPIPPI This is an Android message asking you to select which app to use to scan NFC. If you don't want it to be xDrip+ just disable NFC in xDrip+ settings.

**Comment by ANDREAPIPPI** [comment]
Thanks to both, I didn't suppose it could be an Android message.
